DOC: Documentation fixes

madphysicist opened the pull request:
Typo fixes and text wrappings. More to come as I keep reading the docs.

madphysicist committed:
DOC: Updates to documentation from perusing it in detail.

[ci skip]

madphysicist committed:
MAINT: Wrapped some docstrings and fixed typo

charris commented on doc/source/reference/c-api.types-and-structures.rst:
```diff
@@ -871,8 +871,8 @@ PyArrayIter_Type
-       to its beginning. Note that *backstrides* [k]= *strides* [k]*d
-       *ims_m1* [k], but it is stored here as an optimization.
+       to its beginning. Note that `backstrides`[k]= `strides`[k] *
+       `dims_m1`[k], but it is stored here as an optimization.
```
This doesn't render correctly. You can put single backticks around the whole expression, possibly with a `:code:` preceding, e.g.
```
       How many bytes needed to jump from the end of a dimension back
       to its beginning. Note that :code:`backstrides[k]= strides[k] *
       dims_m1[k]`, but it is stored here as an optimization.
```
You can check things by changing to the `doc` directory, doing `make html` and then viewing `build/html/index.html` in a browser.

njsmith replied:
Note also for future reference that bare single backticks are basically never right in vanilla Sphinx. They assign some text the "default role". If a particular build has been configured to set this to something useful, and you know what that is and it's local convention to use it, then great, but otherwise it's better to be explicit :-). (This is definitely the most annoying incompatibility between markdown and rst.)

charris commented on numpy/core/fromnumeric.py:
```diff
@@ -560,11 +561,12 @@ def partition(a, kth, axis=-1, kind='introselect', order=None):
-    this element and all equal or greater are moved behind it. The ordering of
-    the elements in the two partitions is undefined.
+    Creates a copy of the array with its elements rearranged in such a
+    way that the value of the element in kth position is in the position
```
Should be `k-th` here and elsewhere in the paragraph.

charris commented on numpy/core/fromnumeric.py:
```diff
@@ -573,23 +575,23 @@ def partition(a, kth, axis=-1, kind='introselect', order=None):
-        The order all elements in the partitions is undefined.
-        If provided with a sequence of kth it will partition all elements
-        indexed by kth  of them into their sorted position at once.
+        Element index to partition by. The kth value of the element will
```
See above about `k-th`.

charris commented on numpy/core/fromnumeric.py:
```diff
@@ -658,29 +661,29 @@ def argpartition(a, kth, axis=-1, kind='introselect', order=None):
-        The order all elements in the partitions is undefined.
-        If provided with a sequence of kth it will partition all of them into
-        their sorted position at once.
+        Element index to partition by. The kth element will be in its
```
Again `k-th`.

charris replied:
Might just do a search and replace ;)

madphysicist commented:
Updated all the k-ths and the broken code block. Thanks for letting me know about how to view the docs. My knowledge of sphinx had been purely based on reading the docs and making changes by ear to things that were obviously not showing up correctly.

I made two additional changes:

- Added `_static` folder with empty `.gitignore` to `doc/source` to remove sphinx warning.
- Added `AllOnes` to doc/source/reference/c-api.types-and-structures.rst as a followup to https://github.com/numpy/numpy/pull/7373

charris commented:
What was the warning about `_static` folder?

madphysicist commented:
That it was missing:

    copying static files... WARNING: html_static_path entry '.../numpy/doc/source/_static' does not exist

madphysicist commented:
A better solution might be to create the path in the makefile or in the sphinx setup file.

charris commented:
I suppose there is a possible use of the `_static` file, although I am not clear on what it is for.

madphysicist commented:
The `_static` folder is for customizing the CSS and common images on the site. If the folder is not present in `source`, sphinx generates the default content to `build/_static`.

charris commented on numpy/lib/function_base.py:
```diff
@@ -200,13 +202,14 @@ def _hist_bin_rice(x):
     Returns
     -------
-    n : An estimate of the optimal bin count for the given data.
+    w : An estimate of the optimal bin width for the given data.
```
This is an actual code change. What is the reason here (and below)?

charris commented on numpy/lib/function_base.py:
```diff
@@ -283,46 +285,42 @@ def _hist_bin_fd(x):
 def _hist_bin_auto(x):
     """
-    Histogram bin estimator that uses the maximum of the
+    Histogram bin estimator that uses the minimum width of the
```
Behavior change?

charris commented:
I'd rather the histogram changes were in a separate PR.

madphysicist commented:
They are (#7416). This is a big mistake on my part. Got careless with Git.

charris commented:
I notice that the `doc/build/html/_static` folder is not empty. If we add the folder will the contents still be generated?

madphysicist commented:
Yes. I reran `make html` and everything is fine, including the contents of `build/html/_static`. The `.gitignore` file I added does not interfere with anything.

madphysicist commented:
Removed the histogram stuff.

madphysicist commented:
To be more specific:

Running (from `doc`) 

    rm -rf source/_static
    make html
    cp -r build/html/_static ~/static1

And then

    make clean
    git checkout source/_static/.gitignore # Using git in this case to restore the
                                           # two lines of comment in the gitignore file
    make html
    cp -r build/html/_static ~/static2

Yields the exact same contents for both folders via `diff -r ~/static*`. The `.gitignore` file is not copied over. It just makes it possible to add the missing folder to git so that the warning goes away.

charris commented:
Thanks @madphysicist .
